internal/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,230 @@
+	return "TestOperator allows to run tests with many fake containers"
+}
+
+func (t *TestOperator) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (t *TestOperator) ParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (t *TestOperator) Dependencies() []string {
+	return nil
+}
+
+func (t *TestOperator) CanOperateOn(desc gadgets.GadgetDesc) bool {
+	// Create a new dummy instance
+	dummyInstance, err := desc.(gadgets.GadgetInstantiate).NewInstance()
+	if err != nil {
+		return false
+	}
+
+	// Check whether gadget supports attaching
+	_, ok := dummyInstance.(Attacher)
+	return ok
+}
+
+func (t *TestOperator) Init(params *params.Params) error {
+	return nil
+}
+
+func (t *TestOperator) Close() error {
+	return nil
+}
+
+func (t *TestOperator) Instantiate(gadgetContext operators.GadgetContext, gadgetInstance any, params *params.Params) (operators.OperatorInstance, error) {
+	return &TestOperatorInstance{
+		TestOperator: t, // for Name()
+		attacher:     gadgetInstance.(Attacher),
+		containers:   gadgetContext.Context().Value(TestOperatorContext("containers")).([]*containercollection.Container),
+		testingTB:    gadgetContext.Context().Value(TestOperatorContext("testing.TB")).(testing.TB),
+	}, nil
+}
+
+type TestOperatorInstance struct {
+	*TestOperator
+	attacher   Attacher
+	containers []*containercollection.Container
+	testingTB  testing.TB
+}
+
+func (t *TestOperatorInstance) PreGadgetRun() error {
+	for i := range t.containers {
+		t.attacher.AttachContainer(t.containers[i])
+	}
+	return nil
+}
+
+func (t *TestOperatorInstance) PostGadgetRun() error {
+	for i := range t.containers {
+		t.attacher.DetachContainer(t.containers[i])
+	}
+	return nil
+}
+
+func (t *TestOperatorInstance) EnrichEvent(ev any) error {
+	return nil
+}
+
+func init() {
+	operators.Register(&TestOperatorInstance{})
+}
+
+func BenchmarkAllGadgetsManyContainers(b *testing.B) {
+	utilstest.RequireRoot(b)
+
+	const containerCount = 100
+
+	runtime := &local.Runtime{}
+
+	allGadgets := gadgetregistry.GetAll()
+
+	// Benchmark gadgets in a deterministic order
+	sort.Slice(allGadgets, func(i, j int) bool {
+		a := fmt.Sprintf("%s-%s", allGadgets[i].Category(), allGadgets[i].Name())
+		b := fmt.Sprintf("%s-%s", allGadgets[j].Category(), allGadgets[j].Name())
+		return a < b
+	})
+
+	for _, gadgetDesc := range allGadgets {
+		if _, ok := gadgetDesc.(gadgets.GadgetInstantiate); !ok {
+			continue
+		}
+		categoryAndName := fmt.Sprintf("%s-%s", gadgetDesc.Category(), gadgetDesc.Name())
+		b.Run(categoryAndName, func(b *testing.B) {
+			// Prepare fake containers
+			runnerConfig := &utilstest.RunnerConfig{}
+			var containers []*containercollection.Container
+			for i := 0; i < containerCount; i++ {
+				runner := utilstest.NewRunnerWithTest(b, runnerConfig)
+				container := &containercollection.Container{
+					Netns: runner.Info.NetworkNsID,
+					Pid:   uint32(runner.Info.Tid),
+				}
+				containers = append(containers, container)
+			}
```

Comment:
I'm wondering if this can be moved above to reuse the same fake containers for all gadgets.